src/main/java/com/styra/opa/utils/OPALatencyMeasuringHTTPClient.java

```diff
@@ -0,0 +1,92 @@
+package com.styra.opa.utils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.text.MessageFormat;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static java.util.logging.Level.FINE;
+
+/**
+ * This HTTPClient implementation wraps OPAHTTPClient. In addition to
+ * (optionally) injecting extra headers into the requests, it can also send log
+ * messages to a java.util.logging.Logger instances with latency measurements.
+ */
+public class OPALatencyMeasuringHTTPClient extends OPAHTTPClient {
+
+    private Logger logger;
+
+    private String latencyMeasurementFormatString = "path=''{1}'' latency={0,number,#}ms";
+
+    private MessageFormat fmt;
+
+    private Level latencyMeasurementLogLevel = FINE;
+
+    public OPALatencyMeasuringHTTPClient(Logger logger) {
```

Comment:
It's not normal to pass the logger into the constructor, define the logger in the class with the class name as the name of the logger

Reply:
Each class has it's own logger, no need to change it. Loggers are controlled by central configuration. A normal configuration might be something like com.styra.opa=trace (which would turn on trace logging for all loggers that start with that package), this is why we use the class name as the logger name. You can change the log level for a single class or an entire package, gives you very fine control of logging.

Reply:
I did look into this, and you are totally correct. The way I was implementing the logging stuff was not the conventional way this is done in Java land. I couldn't find any other examples of it being done the way I had it.

Consequently, I have adopted your suggested approach, which aligns with what I have seen done elsewhere in other Java projects.

9a5959928669e504b5945e64e9485a4be013ea19